rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala

```diff
@@ -176,41 +152,38 @@ class NameVarVisitSpec extends FlatSpec with Matchers {
     result.chan should be (ChanVar(FreeVar(0)))
     result.knownFree shouldEqual
         (inputs.knownFree.newBindings(
-            List(("x", NameSort)))._1)
+            List((Some("x"), NameSort)))._1)
   }
   "NameVar" should "Not compile if it's in env of the wrong sort" in {
-    val boundinputs = inputs.copy(env =
-      inputs.env.newBindings(List(("x", ProcSort)))._1)
+    val boundInputs = inputs.copy(env =
+      inputs.env.newBindings(List((Some("x"), ProcSort)))._1)
     
     an [Error] should be thrownBy {
-      nvar.accept(nameVisitor, boundinputs)
+      nvar.accept(nameVisitor, boundInputs)
     }
   }
   "NameVar" should "Not compile if it's used free somewhere else" in {
-    val boundinputs = inputs.copy(knownFree =
```

Comment:
And this says

`for (x <- y) P | new x in Q`

is invalid?

Reply:
x is not free in either of those positions. It's a binder in both.

Reply:
This is saying that the following should not compile:
match P with {
  { x | x } => Q
  _ => R
}

In the pattern, x is free and used twice.

Reply:
I see, thanks.